kong/clustering/init.lua

```diff
@@ -205,14 +205,34 @@ local function fill_empty_hashes(hashes)
   end
 end
 
-function _M:request_version_negotiation()
+
+function _M:call_control_plane(premature)
+  if premature then
+    return
+  end
+
   local response_data, err = version_negotiation.request_version_handshake(self.conf, self.cert, self.cert_key)
   if not response_data then
-    ngx_log(ngx_ERR, _log_prefix, "error while requesting version negotiation: " .. err)
-    assert(ngx.timer.at(math.random(5, 10), function(premature)
-      self:communicate(premature)
-    end))
-    return
+    ngx_log(ngx_ERR, _log_prefix, "error while requesting version negotiation: ", err)
+    ngx_log(ngx_ERR, _log_prefix, "Assuming old JSON protocol.")
+  end
+
+  local config_proto, msg = version_negotiation.get_negotiated_service("config")
```

Comment:
Should we still go though this if `request_version_handshake` fails? What do we get in that case?

Reply:
And if we already determined to use old protocol, we should not see the "error reading negotiated..." log.